### PR TITLE
Feat/activity SLIDER item contract

### PR DESCRIPTION
**Pull request by vmkhitaryanscn**

### CODE REVIEW

File: src/shared/ui/survey/Slider/SurveySlider.tsx [hunk 1]
```diff
@@ -10,45 +10,44 @@ const SurveySlider: FC<SliderProps> = ({ config, ...props }) => {
...
 
-  const initialIndex = items.findIndex(({ value }) => value === initialValue);
+  const length = maxValue - minValue;
+  const items = new Array(length).fill(1);
 
   return (
     <YStack>
       <Slider
-        defaultValue={[initialIndex !== -1 ? initialIndex : -100]}
+        defaultValue={[initialValue]}
```
Comment (anq83) [inline src/shared/ui/survey/Slider/SurveySlider.tsx:35]: If you set min-max in the test dto lets say 10-20
then slider is absend when start activity

File: src/shared/ui/survey/Slider/SurveySlider.tsx [hunk 2]
```diff
@@ -10,45 +10,44 @@ const SurveySlider: FC<SliderProps> = ({ config, ...props }) => {
...
         size={THUMB_SIZE}
-        max={items.length - 1}
+        max={maxValue - 1}
+        min={minValue}
         step={isContinuousSlider ? 0.01 : 1}
       />
 
       <XStack jc="space-between" mt={9}>
-        {items.map(({ value }) => {
+        {items.map((value, index) => {
```
Comment (anq83) [inline src/shared/ui/survey/Slider/SurveySlider.tsx:46]: Please fill necessary values into items array (i.o 1) and use value, remove index

**Review by anq83 — CHANGES_REQUESTED** [review]

**Review by anq83 — APPROVED** [review]

File: src/features/pass-survey/ui/ActivityItem.tsx [hunk 3]
```diff
@@ -56,6 +57,20 @@ function ActivityItem({ value, pipelineItem, onResponse }: Props) {
       );
       break;
 
+    case 'Slider':
+      item = (
+        <Box flex={1} jc="center" mx="$5">
+          <SurveySlider
+            config={pipelineItem.payload}
+            onChange={() => console.log('changed')}
```
Comment (BamMironov) [inline src/features/pass-survey/ui/ActivityItem.tsx:65]: Add onResponse to onChange prop

File: src/shared/ui/survey/Slider/SurveySlider.tsx [hunk 4]
```diff
@@ -10,41 +10,42 @@ const SurveySlider: FC<SliderProps> = ({ config, ...props }) => {
...
-    const roundedValue = Math.round(numericValue * 100) / 100;
+    const roundedValue = Math.round(value * 100) / 100;
     onChange(roundedValue);
   };
 
-  const initialIndex = items.findIndex(({ value }) => value === initialValue);
+  const items = [];
+  for (let i = minValue; i <= maxValue; i++) {
+    items.push(i);
+  }
```
Comment (BamMironov) [inline src/shared/ui/survey/Slider/SurveySlider.tsx:32]: Please use `Array.from` instead. Something  like this:
```
Array.from(
    { length: maxValue - minValue },
    (_, index) => index + minValue,
 )
```

**Review by BamMironov — CHANGES_REQUESTED** [review]

**Review by BamMironov — APPROVED** [review]

**Pull request merged**